sycl/doc/design/SYCL-1Pass.md

```diff
@@ -0,0 +1,57 @@
+# SYCL 1-Pass Compilation Model
+
+Currently the DPC++ compiler utilizes a scheme which needs multiple invocations of the compiler.  Specifically, it invokes the cc1 compiler once for each device compilation and one more time for the host compilation.  This document describes design changes necessary (in the various components of the DPC++ Toolchain) to switch to a single invocation of the cc1 compiler for each Translation Unit (TU).
+
+## Compiler Front End (FE) Changes
+
+This design reuqires some changes from the FE.  In particular, it needs to emit code via the normal/standard C++ rules, and in addition, needs to emit functions that are required for Kernel code, which may not have been emitted via the normal C++ rules.
+
+### Generation and dispatch to the OpenCL Kernel
+
+The FE will generate the OpenCL Kernel which creates the Functor object that it will call into.  FE then replaces the [[sycl_kernel]] function with one that dispatches to the OpenCL Kernel.  New APIs from the SYCL Runtime Library (RTL) are needed to accomplish this task.
+
+
+### Elimination of generation of the integration header file
+
+Since there is no longer separate device and host compilation invocations, communication between them via the integration header file is no longer necessary.  Instead the FE can pass the information previously generated in the header file directly to the RTL via new APIs.
```

Comment:
Based on today's discussion, this likely still needs to stick around, but only be enabled 'sometimes'.